### ISSUE TITLE
Trimming removes all clickable links in the PDF viewer

### ISSUE THREAD

**Issue by stepelu** [James-Yu/LaTeX-Workshop#3502]
### Disable all the other extensions except for LaTeX Workshop, restart VS Code, and check that you still see this issue. [Required]

You still see this issue?: Yes

### Make sure to visit the [wiki FAQ](https://github.com/James-Yu/LaTeX-Workshop/wiki/FAQ) before filling an issue.

You visited the wiki?: Yes

## Describe the bug [Required]

Consider the following example:
```
\documentclass{article}
\usepackage{hyperref}

\begin{document}
\begin{equation}\label{eq:test}
dX_t = f(X_t)dt
\end{equation}
This is equation \ref{eq:test}.
\end{document}
```

It results on the following rendering, the red box is clickable (as it should be):
<img width="479" alt="image" src="https://user-images.githubusercontent.com/6617696/194829712-9f50d064-f8c8-4117-be7c-7dce5ca0be03.png">

Setting page trimming to any positive (here 10%) in the PDF viewer results in the following rendering:
<img width="598" alt="image" src="https://user-images.githubusercontent.com/6617696/194829953-6598e4c3-9236-414a-9312-09445cb7bfff.png">
The red box is gone and it is no longer possible to click on 1 to redirect the view to equation 1.

### To Reproduce

1. Use the provided latex code
2. Compile and open view with included PDF viewer
3. Set page trimming to 10%

### Expected behavior

Trimming should not affect links rendering and clickability.

### LaTeX Workshop Output [Required]

```
[17:57:36] Initializing LaTeX Workshop.
[17:57:36] Extension root: /Users/sp/.vscode/extensions/james-yu.latex-workshop-8.29.0
[17:57:36] $PATH: /opt/homebrew/Cellar/pyenv-virtualenv/1.1.5/shims:/Users/sp/.pyenv/shims:/opt/homebrew/bin:/opt/homebrew/sbin:/Users/sp/Bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/Library/TeX/texbin
[17:57:36] $SHELL: /opt/homebrew/bin/fish
[17:57:36] $LANG: undefined
[17:57:36] $LC_ALL: undefined
[17:57:36] process.platform: darwin
[17:57:36] process.arch: arm64
[17:57:36] vscode.env.appName: Visual Studio Code
[17:57:36] vscode.env.remoteName: undefined
[17:57:36] vscode.env.uiKind: 1
[17:57:36] Configuration for workspace: file:///Users/sp/Latex/fwd-diffusion-model
[17:57:36] editor.acceptSuggestionOnEnter: "on"
[17:57:36] latex-workshop.bind.enter.key: true
[17:57:36] latex-workshop.docker.enabled: false
[17:57:36] latex-workshop.docker.image.latex: ""
[17:57:36] latex-workshop.hover.preview.mathjax.extensions: []
[17:57:36] latex-workshop.intellisense.package.enabled: true
[17:57:36] latex-workshop.intellisense.update.aggressive.enabled: false
[17:57:36] latex-workshop.intellisense.update.delay: 1000
[17:57:36] latex-workshop.latex.autoBuild.run: "onFileChange"
[17:57:36] latex-workshop.latex.build.forceRecipeUsage: true
[17:57:36] latex-workshop.latex.outDir: "%DIR%"
[17:57:36] latex-workshop.latex.recipes: [
 {
  "name": "latexmk 🔃",
  "tools": [
   "latexmk"
  ]
 },
 {
  "name": "latexmk (latexmkrc)",
  "tools": [
   "latexmk_rconly"
  ]
 },
 {
  "name": "latexmk (lualatex)",
  "tools": [
   "lualatexmk"
  ]
 },
 {
  "name": "latexmk (xelatex)",
  "tools": [
   "xelatexmk"
  ]
 },
 {
  "name": "pdflatex ➞ bibtex ➞ pdflatex × 2",
  "tools": [
   "pdflatex",
   "bibtex",
   "pdflatex",
   "pdflatex"
  ]
 },
 {
  "name": "Compile Rnw files",
  "tools": [
   "rnw2tex",
   "latexmk"
  ]
 },
 {
  "name": "Compile Jnw files",
  "tools": [
   "jnw2tex",
   "latexmk"
  ]
 },
 {
  "name": "tectonic",
  "tools": [
   "tectonic"
  ]
 }
]
[17:57:36] latex-workshop.latex.tools: [
 {
  "name": "latexmk",
  "command": "latexmk",
  "args": [
   "-synctex=1",
   "-interaction=nonstopmode",
   "-file-line-error",
   "-pdf",
   "-outdir=%OUTDIR%",
   "%DOC%"
  ],
  "env": {}
 },
 {
  "name": "lualatexmk",
  "command": "latexmk",
  "args": [
   "-synctex=1",
   "-interaction=nonstopmode",
   "-file-line-error",
   "-lualatex",
   "-outdir=%OUTDIR%",
   "%DOC%"
  ],
  "env": {}
 },
 {
  "name": "xelatexmk",
  "command": "latexmk",
  "args": [
   "-synctex=1",
   "-interaction=nonstopmode",
   "-file-line-error",
   "-xelatex",
   "-outdir=%OUTDIR%",
   "%DOC%"
  ],
  "env": {}
 },
 {
  "name": "latexmk_rconly",
  "command": "latexmk",
  "args": [
   "%DOC%"
  ],
  "env": {}
 },
 {
  "name": "pdflatex",
  "command": "pdflatex",
  "args": [
   "-synctex=1",
   "-interaction=nonstopmode",
   "-file-line-error",
   "%DOC%"
  ],
  "env": {}
 },
 {
  "name": "bibtex",
  "command": "bibtex",
  "args": [
   "%DOCFILE%"
  ],
  "env": {}
 },
 {
  "name": "rnw2tex",
  "command": "Rscript",
  "args": [
   "-e",
   "knitr::opts_knit$set(concordance = TRUE); knitr::knit('%DOCFILE_EXT%')"
  ],
  "env": {}
 },
 {
  "name": "jnw2tex",
  "command": "julia",
  "args": [
   "-e",
   "using Weave; weave(\"%DOC_EXT%\", doctype=\"tex\")"
  ],
  "env": {}
 },
 {
  "name": "jnw2texmintex",
  "command": "julia",
  "args": [
   "-e",
   "using Weave; weave(\"%DOC_EXT%\", doctype=\"texminted\")"
  ],
  "env": {}
 },
 {
  "name": "tectonic",
  "command": "tectonic",
  "args": [
   "--synctex",
   "--keep-logs",
   "%DOC%.tex"
  ],
  "env": {}
 }
]
[17:57:36] latex-workshop.viewer.pdf.internal.keyboardEvent: "auto"
[17:57:36] Creating a new file watcher.
[17:57:36] watcherOptions: {"useFsEvents":false,"usePolling":false,"interval":300,"binaryInterval":1000,"awaitWriteFinish":{"stabilityThreshold":250}}
[17:57:36] Creating PDF file watcher.
[17:57:36] watcherOptions: {"useFsEvents":false,"usePolling":false,"interval":300,"binaryInterval":1000,"awaitWriteFinish":{}}
[17:57:36] Creating Bib file watcher.
[17:57:36] watcherOptions: {"useFsEvents":false,"usePolling":false,"interval":300,"binaryInterval":1000,"awaitWriteFinish":{"stabilityThreshold":250}}
[17:57:36] Set $LATEXWORKSHOP_DOCKER_LATEX: ""
[17:57:36] [Server] Creating LaTeX Workshop http and websocket server.
[17:57:36] LaTeX Workshop initialized.
[17:57:36] Trigger characters for intellisense of LaTeX documents: ["\\",",","{"]
[17:57:36] Bibtex format config: {"tab":"  ","case":"lowercase","left":"{","right":"}","trailingComma":false,"sort":["key"],"alignOnEqual":true,"sortFields":true,"fieldsOrder":[],"firstEntries":["string","xdata"]}
[17:57:36] Current workspace folders: ["file:///Users/sp/Latex/fwd-diffusion-model"]
[17:57:36] Found root file from active editor: /Users/sp/Latex/fwd-diffusion-model/test.tex
[17:57:37] Root file changed: from undefined to /Users/sp/Latex/fwd-diffusion-model/test.tex
[17:57:37] Start to find all dependencies.
[17:57:37] Root file languageId: latex
[17:57:37] [Server] Server successfully started: {"address":"127.0.0.1","family":"IPv4","port":59691}
[17:57:37] Reset file watcher.
[17:57:37] Parsing a file and its subfiles: /Users/sp/Latex/fwd-diffusion-model/test.tex
[17:57:37] Parse fls file.
[17:57:37] Fls file found: /Users/sp/Latex/fwd-diffusion-model/test.fls
[17:57:37] Parse aux file: /Users/sp/Latex/fwd-diffusion-model/test.aux
[17:57:37] Linter for root file started.
[17:57:37] Linter for root command: chktex
[17:57:37] Linter for root command args: ["-wall","-n22","-n30","-e16","-q","-n21","-n3","-n9","-n17","-f%f:%l:%c:%d:%k:%n:%m\n","/Users/sp/Latex/fwd-diffusion-model/test.tex"]
[17:57:37] Added to file watcher: /Users/sp/Latex/fwd-diffusion-model/test.tex
[17:57:37] Snippet data loaded.
[17:57:37] Checking for duplicate labels: /Users/sp/Latex/fwd-diffusion-model/test.tex.
[17:57:37] [Server] valdOrigin is http://127.0.0.1:59691
[17:57:37] Restoring the PDF viewer at the column 1 from the state: {"type":"state","state":{"pdfFileUri":"file:///Users/sp/Latex/fwd-diffusion-model/test.pdf","scale":"1.2408088235294117","scrollMode":0,"spreadMode":0,"scrollTop":0,"scrollLeft":0,"trim":2,"synctexEnabled":true,"autoReloadEnabled":true}}
[17:57:37] Linter for root successfully finished in 0s 74ms
[17:57:37] The .chktexrc file not found.
[17:57:37] Linter log parsed with 1 messages.
[17:57:37] The internal PDF viewer url: http://127.0.0.1:59691/viewer.html?file=pdf..ZmlsZSUzQSUyRiUyRiUyRlVzZXJzJTJGc3AlMkZMYXRleCUyRmZ3ZC1kaWZmdXNpb24tbW9kZWwlMkZ0ZXN0LnBkZg
[17:57:37] Added to PDF file watcher: file:///Users/sp/Latex/fwd-diffusion-model/test.pdf
[17:57:37] Handle data type: open
[17:57:37] Preview PDF file: file:///Users/sp/Latex/fwd-diffusion-model/test.pdf
[17:57:37] Handle data type: loaded
[17:57:38] BUILD command invoked.
[17:57:38] The document of the active editor: file:///Users/sp/Latex/fwd-diffusion-model/test.tex
[17:57:38] The languageId of the document: latex
[17:57:38] Current workspace folders: ["file:///Users/sp/Latex/fwd-diffusion-model"]
[17:57:38] Found root file from active editor: /Users/sp/Latex/fwd-diffusion-model/test.tex
[17:57:38] Keep using the same root file: /Users/sp/Latex/fwd-diffusion-model/test.tex
[17:57:38] Building root file: /Users/sp/Latex/fwd-diffusion-model/test.tex
[17:57:38] Build root file /Users/sp/Latex/fwd-diffusion-model/test.tex
[17:57:38] outDir: /Users/sp/Latex/fwd-diffusion-model
[17:57:38] Recipe step 1: latexmk
[17:57:38] Recipe step 1 args: ["-synctex=1","-interaction=nonstopmode","-file-line-error","-pdf","-outdir=/Users/sp/Latex/fwd-diffusion-model","/Users/sp/Latex/fwd-diffusion-model/test"]
[17:57:38] Recipe step env: {}
[17:57:38] cwd: /Users/sp/Latex/fwd-diffusion-model
[17:57:38] LaTeX build process spawned. PID: 98109.
[17:57:39] Recipe of length 1 finished. PID: 98109.
[17:57:39] Successfully built /Users/sp/Latex/fwd-diffusion-model/test.tex.
[17:57:39] Manager.fileWatcher.getWatched: {"/Users/sp/Latex/fwd-diffusion-model":["test.tex"]}
[17:57:39] Manager.filesWatched: ["/Users/sp/Latex/fwd-diffusion-model/test.tex"]
[17:57:39] BibWatcher.bibWatcher.getWatched: {}
[17:57:39] BibWatcher.bibsWatched: []
[17:57:39] PdfWatcher.pdfWatcher.getWatched: {"/Users/sp/Latex/fwd-diffusion-model":["test.pdf"]}
[17:57:39] PdfWatcher.pdfsWatched: ["/Users/sp/Latex/fwd-diffusion-model/test.pdf"]
[17:57:39] PdfWatcher.watchedPdfVirtualUris: []
[17:57:39] PdfWatcher.ignoredPdfUris: ["file:///Users/sp/Latex/fwd-diffusion-model/test.pdf"]
```

## Desktop [Required]

Please write exact version numbers. Please don't write `latest` instead of exact numbers.

 - OS: OSX 12.6
 - VS Code version: Version: 1.72.0 (Universal)
 - Extension version: [8.29.0] - 2022-08-25
 - TeX distribution version: Tex Live 20220321 (HomeBrew installation)

### Are you using VSCodium?

No

### Are you using the Snap or Flatpack versions of VS Code?

No

### Are you using LaTeX Workshop with VS Code Remote?

No

**Comment by James-Yu** [comment]
Unfortunately, I cannot reproduce this issue. Can you please provide a minimal working example and your generated PDF?

**Comment by stepelu** [comment]
I am including here all the files generated by the snipped above that exhibits the issue.

[Archive.zip](https://github.com/James-Yu/LaTeX-Workshop/files/9754098/Archive.zip)

Also, maybe this further example can shed some light (the issue seems to be with the PDF.js viewer, not with the way the pdf has been generated, and exclusively affect trimming, not zooming).

This is what the https://arxiv.org/abs/2210.01554 pdf looks like by simply opening it in vscode, without trimming:

<img width="593" alt="image" src="https://user-images.githubusercontent.com/6617696/195036257-6eb81a47-cb55-4585-800d-a8b607b7464e.png">

and with 5% trimming:

<img width="747" alt="image" src="https://user-images.githubusercontent.com/6617696/195036327-e782e13b-6a31-4a7d-a6c4-accf2637b53d.png">

It looks like the coordinate system for the links gets broken by trimming, everything is shifted to the left.

I don't remember this issue being there in the past.
In case you are on OSX as well I am not sure why it does not reproduce, I'm using an M1 CPU if that makes any difference.

**Comment by James-Yu** [comment]
<img width="941" alt="Screen Shot 2022-10-11 at 21 08 42" src="https://user-images.githubusercontent.com/4210342/195099476-7f91677f-b0e2-4ed0-85e4-9771c52f7880.png">

trim 5% on my side, M1 also. I'm afraid this is a local issue. You may consider uninstall extension, exit & open vscode, reinstall and see if issue resolved. Unless reproducible on the dev side, I cannot fix the potential bug.

**Comment by stepelu** [comment]
@James-Yu thank you for the suggestion, indeed uninstalling and reinstalling the extension solved the problem.
As mentioned on issue #3478, the provided vsix file https://github.com/James-Yu/LaTeX-Workshop/issues/3487#issuecomment-1257148177 did not enable the feature for me, but somehow introduced the issue I was seeing here.
Glad all is solved and working now!

**Comment by stepelu** [comment]
@James-Yu the issue re-appeared for me with the latest version of the extension.
I went as far as removing vscode, removing all settings as in https://stackoverflow.com/a/47561948, and reinstalling vscode.
The issue is still present for me, could you please check again whether the issue is present at this time with you, or suggest me a way forward to help debugging this issue?

**Comment by James-Yu** [comment]
I can reproduce now. Will look into it. However, I am a bit pessimistic on resolving the issue, as I am not familiar with the trimming.

**Comment by James-Yu** [comment]
Have to admit that the trimming code is quite messy. Will return to refactor later.

**Comment by scaomath** [comment]
Related: #3559
Interestingly I did not realize that it is the clickable boxes having been moved because I always have `colorlinks=true` in my `\hypersetup` 
Now locally this issue has been resolved for me after 9.1.0.
